File: astraquant-backend/services/exchange-access-gateway/internal/domain/models/instrument.go
package models

type UnifiedInstrument struct {
	Exchange       string `json:"exchange"`
	InternalSymbol string `json:"internal_symbol"`
	ExchangeSymbol string `json:"exchange_symbol"`
	BaseAsset      string `json:"base_asset"`
	QuoteAsset     string `json:"quote_asset"`
	MarginAsset    string `json:"margin_asset"`
	ContractType   string `json:"contract_type"`
	TickSize       string `json:"tick_size"`
	LotSize        string `json:"lot_size"`
	MinSize        string `json:"min_size"`
	ContractValue  string `json:"contract_value"`
	PricePrecision int    `json:"price_precision"`
	SizePrecision  int    `json:"size_precision"`
	Status         string `json:"status"`
}

type GetInstrumentsRequest struct {
	ContractType string
	Symbol       string
}

File: astraquant-backend/services/market-data-service/app/schemas/instrument.py
from pydantic import BaseModel

class InstrumentResponse(BaseModel):
    exchange: str
    internal_symbol: str
    exchange_symbol: str
    base_asset: str | None = None
    quote_asset: str | None = None
    margin_asset: str | None = None
    contract_type: str
    tick_size: str | None = None
    lot_size: str | None = None
    min_size: str | None = None
    contract_value: str | None = None
    status: str
